librad/src/git/fetch.rs

```diff
@@ -403,8 +504,14 @@ impl<'a> DefaultFetcher<'a> {
 
             let mut callbacks = git2::RemoteCallbacks::new();
             callbacks.transfer_progress(|prog| {
-                tracing::trace!("Fetch: received {} bytes", prog.received_bytes());
-                true
+                let received_bytes = prog.received_bytes();
+                tracing::trace!("Fetch: received {} bytes", received_bytes);
+                if received_bytes < MAX_FETCH {
```

Comment:
> One question that came to mind though is: if a malkovich doesn't update a project for _a long time_ could they potentially be stuck in being unable to update the project because the delta is too big to fetch?

Ya... another reason to have different limits. The `rad/id` branches are not supposed to change very often. The `rad/signed_refs`, otoh, change all the time -- they will also not compress that well, because they contain hashes. Still, **a lot** of updates should fit into a few hundred KiB of packfile. Everything else... well, Linus' kernel tree is something like 2.5G.